src/common/pico_base/include/pico/config.h

```diff
@@ -18,4 +18,9 @@
 
 #include "pico/config_autogen.h"
 
+// PICO_CONFIG: PICO_CONFIG_RTOS_ADAPTER_HEADER, path to header include in the default pico/config.h for RTOS integration defines that must be included in all sources, type=string, default=none, group=pico_base
```

Comment:
Is this definitely specific to RTOS integration, or could it have other uses too?

Reply:
It was intended to be specific (you could use it for something else, but then you have no idea if you're colliding with something else). There is no easy way to make a generic include interface (except for the CMake auto-gen which we already have but are deliberately not using), so I just picked this one out since it is the one I know that is needed now